### PR TITLE
Category validation persist validation

### PR DESCRIPTION
**Pull request by fefrain**

### CODE REVIEW

File: common/survey/_survey/surveyCategories.js [hunk 1]
```diff
@@ -11,6 +13,17 @@ const getCategories = R.pipe(
 const getCategoriesArray = R.pipe(
   getCategories,
   R.values,
+  //sort by name
+  R.sort((category1, category2) => {
+    const name1 = Category.getName(category1)
+    const name2 = Category.getName(category2)
+
```
Comment (minotogna) [inline common/survey/_survey/surveyCategories.js]: should they be both converted to upper case or lower case?

File: common/survey/category.js [hunk 2]
```diff
@@ -45,19 +45,15 @@ const newCategory = (props = {}) => {
...
-
-  return {
-    [CategoryLevel.keys.uuid]: uuidv4(),
-    [CategoryLevel.keys.categoryUuid]: ObjectUtils.getUuid(category),
-    [CategoryLevel.keys.index]: index,
-    [CategoryLevel.keys.props]: {
-      [CategoryLevel.props.name]: 'level_' + (index + 1),
-      ...props
-    }
+const newLevel = (category, props = {}, index = getLevelsArray(category).length) => ({
```
Comment (minotogna) [inline common/survey/category.js]: why not to use ```Object.keys(Category.getLevels(category)).length```

File: server/modules/survey/service/publish/jobs/categoriesValidationJob.js [hunk 3]
```diff
@@ -12,15 +12,16 @@ class CategoriesValidationJob extends Job {
   }
 
   async execute (tx) {
-    const categories = await CategoryManager.fetchCategoriesBySurveyId(this.surveyId, true, false, tx)
+    const categories = await CategoryManager.fetchCategoriesAndLevelsBySurveyId(this.surveyId, true, true, tx)
 
-    this.total = categories.length
+    const categoriesArr = Object.values(categories)
```
Comment (minotogna) [inline server/modules/survey/service/publish/jobs/categoriesValidationJob.js]: why don't you iterate with Object.entries?

File: common/survey/category.js (hunk 2 above)
Comment (fefrain) [reply common/survey/category.js]: right

File: server/modules/survey/service/publish/jobs/categoriesValidationJob.js (hunk 3 above)
Comment (fefrain) [reply server/modules/survey/service/publish/jobs/categoriesValidationJob.js]: you need only the values, not the keys (uuids) of categories

Comment (minotogna) [reply server/modules/survey/service/publish/jobs/categoriesValidationJob.js]: yes, but you first get the values and then iterate. what i meant you can get the entries only. then iterate over them as well as get the length

**Pull request merged**